examples/repository/_simplerepo.py

```diff
@@ -131,3 +135,73 @@ def add_target(self, path: str, content: str) -> None:
         # update snapshot, timestamp
         self.snapshot()
         self.timestamp()
+
+    def submit_delegation(self, rolename: str, data: bytes) -> bool:
+        """Add a delegation to a (offline signed) delegated targets metadata"""
+        try:
+            logger.debug("Processing new delegation to role %s", rolename)
+            keyid, keydict = next(iter(json.loads(data).items()))
+            key = Key.from_dict(keyid, keydict)
+
+            # add delegation and key
+            role = DelegatedRole(rolename, [], 1, True, [f"{rolename}/*"])
+            with self.edit("targets") as targets:
+                if targets.delegations is None:
+                    targets.delegations = Delegations({}, {})
+
+                targets.delegations.roles[rolename] = role
+                targets.add_key(key, rolename)
+
+        except (RepositoryError, json.JSONDecodeError) as e:
```

Comment:
Is it worth catching anything from `keyid, keydict = next(iter(json.loads(data).items()))` here?
AttributeError, StopIteration, ...?

Reply:
... to gracefully fail if the user throws malformed data at the API.

Reply:
Probably? I think I'll leave this for future improvement

Reply:
Yeah, you are right. Again, I was looking at it from the perspective of random users. But it's a demo after all. :)

Reply:
I guess, I was also wondering, why you used `next` and `iter` instead of just popping the first item tuple.